### ISSUE TITLE
GPIO Config 2 : GPIO Keyboard and Joysticks does not work properly

### ISSUE THREAD

**Issue by NurseEdna** [randyrossi/bmc64#268]
Hi, 

the BMC64 works really fine on my Pi3, even attached Joysticks to the GPIO (Config 1). Works well.

Then I removed the Joy-GPIO wiring and attached the C64II keyboard as shown in your corresponding section for GPIO Config 2 and it almost worked.

Unfortunately some keys (highlighted in red below) did not respond at all, some other keys (yellow) responded, but yielded not the correct letter resp. symbol.

So my first thought was that I messed up with one two cables in the breadboard wiring btw. keyboard and GPIO, but I checked that several times against the table in the section for GPIO Config 2 and found nothing wrong.

Do you have any clue what might be the issue here (except for a defective keyboard and bad wiring)?  

Thanks a lot 
Best regards

![keyboard](https://github.com/user-attachments/assets/b496064e-880d-4fc6-a720-dee5a91ef3d0)

**Comment by nickgoodmanuk** [comment]
Have a look at the matrix here : 

https://sta.c64.org/cbm64kbdlay.html

Your red keys are all on bit4 - suggesting a bad connection to a GPIO pin.  Yellow keys on on bit5 - maybe that connection is iffy too.  Hope that helps.

**Comment by NurseEdna** [comment]
Yeah, I also saw the affected keys belong to the same bit, so either the keyboard is broken or (most likely) I messed up with the GPIO wiring.

Thanks for the hint :-) I will give it a try and re-wire the pins.

Regards

**Comment by aminch** [comment]
User wiring issue, closing.
